heron/api/src/java/com/twitter/heron/api/state/HashMapState.java

```diff
@@ -0,0 +1,25 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.api.state;
+
+import java.io.Serializable;
+import java.util.HashMap;
+
+/**
+ * A State backed up by HashMap
+ */
+public class HashMapState extends HashMap<Serializable, Serializable> implements State {
```

Comment:
interface State is already a map. If a hashmap member placed here, it leads to nested map.